### ISSUE TITLE
Trills not working when applied to the later half of tied notes

### ISSUE THREAD

**Issue by sampleeditor** [musescore/MuseScore#24843]
### Issue type

Muse Sounds bug

### Description with steps to reproduce

1. Create a score that uses a Muse Sounds instrument with sampled trills such as Muse Strings Violin 1 (Solo)
2. Tie two whole notes (such as A4) over the first two bars
3. Add a trill line to the second bar over the tied note. 
Actual behaviour: Note sustains but trill does not play.
Expected behaviour: Sustained note sounds and then begins to trill at the start of the second bar with the trill line

### Supporting files, videos and screenshots

https://github.com/user-attachments/assets/75afc4b4-7a04-4d28-87a8-24eadf998e4b

[Violin Trill Over Tie.mscz.zip](https://github.com/user-attachments/files/17079868/Violin.Trill.Over.Tie.mscz.zip)

### What is the latest version of MuseScore Studio where this issue is present?

MuseScore 4.4.2.242530503 Nightly

### Regression

I was unable to check

### Operating system

macOS 13.5

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by sampleeditor** [comment]
Confirmed by @konradglas, this is not a sampler issue nor Muse Sounds issue, but a MuseScore playback issue.

**Comment by wizofaus** [comment]
I already reported this against Muse Strings via Muse Hub. It doesn't happen with MS Basic or CineStrings solo, so I can't see how it's a MuseScore issue..

**Comment by SteveBlower** [comment]
Why would you use a tie rather than a slur in such a case?  The start of the trill would necessitate a player to re-articulate the note.  Why shouldn't Musescore do the same?

**Comment by wizofaus** [comment]
Why does the start of a trill necessitate rearticulation? And at any rate, slurs & ties mean the same thing (generally): don't re-articulate!
A tie clearly means that you effectively start the trill on the upper note immediately after the beat.

**Comment by SteveBlower** [comment]
*"Why does the start of a trill necessitate rearticulation?"*

Because at the start of a trill you stop playing the lower note and then start playing (i.e. articulate) the upper note - and then repeatedly swap between lower and upper.  A tie doesn't have a role between notes of two different pitches

**Comment by wizofaus** [comment]
Since when was playing a new pitch "articulation" (on anything other keyboard instruments anyway)?

**Comment by SteveBlower** [comment]
All right then, you slur on to the upper note, but you certainly don't **tie** on to it. So use a slur to create a slur onto a different pitched note. Don't use a tie , it won't work!

**Comment by wizofaus** [comment]
Did you look at the score I linked to?
It's just a playback bug that it doesn't play trills if they're tied to (affects many of the free muse sounds libraries, but not all).
Not sure why we're discussing slurs at all.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by konradglas** [comment]
It is indeed a playback bug related to the MuseSampler Wrapper. Which is part of the MuseScore code base.
We know about the issue and will provide a fix.

**Comment by SteveBlower** [comment]
My point is that tying onto a trill is illogical and the behaviour you describe is better achieved using a slur.  No fix needed.

**Comment by wizofaus** [comment]
As I said, slurring between two notes of the same pitch makes no sense. Notes ties to trills are not uncommon, it's quite clear what it means.
A slur would make sense if you wanted the trill to start on the upper note, but that would require a grace note or similar to make that clear, and then sure, you could slur to that.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by SteveBlower** [comment]
Slurs between notes of the same pitch are very common.  It is generally interpreted as requiring a softer articulation on the slurred-to note than would be used without a slur. But interpretation is instrument and context dependent.  As a clarinettist I would probably use a "du" rather than a "tu" tonguing action.  A violinist would do something different, perhaps play the two notes without changing bow direction but stopping or slowing the bow between notes.  

But that is all getting away from the topic of the issue and my argument, which is that the desired sound can be achieved if a slur is used from the un-trilled note to the trilled note.  It seems a waste of effort to try to do the same when a tie is used instead.

**Comment by wizofaus** [comment]
Accepted, my statement came across as overly general - specifically referring to the case of just two notes of the same pitch with no other articulation marks over them.
Anyway what we're trying to notate is the following (more or less), which can't be done with just slur:

![image](https://github.com/user-attachments/assets/3a2458a9-d4b2-41ad-a04b-00a0c3f0147d)

**Comment by aliciapaz** [comment]
I have a similar issue. If I have a trill on the first beat of two tied notes, playback stops at the end of the first bar (the red note does not produce any sound). Trill line has the same behavior. 
I'm on MuseScore Studio 4.4.2.242570931-x86_64
![image](https://github.com/user-attachments/assets/609e4eef-c94f-4249-ae71-e44f8ec7217f)

**Comment by BoBogus** [comment]
I know this issue has been reported a thousand times already, but I think the problem might come down to how MuseScore internally handles tied notes.  And a potential solution has occurred to me.

In a real-life, live performance, tied notes are treated by performers as though they were one single note of extra long duration.  But clearly this isn't happening in MuseScore — instead it is interpreting these tied notes as separate durations.  For example, a musician would play two whole notes and a half note tied together as a single note with a duration of 10 beats.  MuseScore instead seems to treat the component notes of the tie as separate notes of 4, 4, and 2 beats, and then applies the tie only at playback itself.  This works fine for sustains, but a bug of some sort keeps it from being properly applied to trills and ties.

So what if we fix that at the root?  Nothing changes from one measure to the next, so why not just treat tied notes as one single note?  If you want a trill or tremolo to carry over to another note of the same pitch, you can just use a slur since it looks the same as a tie.

To summarize, if the code could be reworked so that MuseScore immediately sums the component durations of tied notes and then uses only this combined value in its later calculations and export to playback, wouldn't this solve the issue?
